ddp-workspace/projects/ddp-sdk/src/lib/services/fileUpload.service.ts

```diff
@@ -21,23 +21,24 @@ export class FileUploadService extends UserServiceAgent<any> {
         super(session, configuration, http, logger, null);
     }
 
-    getUploadUrl(studyGuid: string, activityGuid: string, questionStableId: string, file: File): Observable<FileUploadResponse> {
+    getUploadUrl(studyGuid: string, activityGuid: string, questionStableId: string, files: File[]): Observable<FileUploadResponse[]> {
         const path = `/studies/${studyGuid}/activities/${activityGuid}/uploads`;
 
-        const requestBody: FileUploadBody = {
-            questionStableId,
-            fileName: file.name,
-            fileSize: file.size,
-            mimeType: file.type
-        };
-
-        return this.postObservable(path, requestBody, {}, true).pipe(
-            catchError(error => {
-                this.logger.logDebug('getUploadUrl error', error);
-                return throwError(error.error);
+        const acceptedFiles: Observable<FileUploadResponse | null>[] = files.map<FileUploadBody>(file =>  ({
```

Comment:
Have one **map** would be enough I believe.
Like 

```
files.map( file => {
  const accFile = {...};
  this.postObservable(path, accFile, {}, true) ... // and so one
})
```

Reply:
Working.

Reply:
Do you think this implementation is fine? 
![image](https://user-images.githubusercontent.com/77500504/160658390-27a05d11-3c4d-4d8e-8872-b253568f06ec.png)

Reply:
Yes, it looks better imho.

And I suppose 
`files.map<Observable<FileUploadResponse>>` can be just `files.map(...)` as the type is already defined in `acceptedFiles$`.

Reply:
Okay. will be done! Thanks